drake/systems/plants/constraint/PostureChangeConstraint.m

```diff
@@ -37,20 +35,17 @@
       obj.ub_change = ub_change;
     end
   end
-  
+
   methods
-    function obj = PostureChangeConstraint(robot,joint_ind,lb_change,ub_change,tspan)
-      if(nargin<5)
-        tspan = [-inf inf];
-      end
-      ptr = constructPtrRigidBodyConstraintmex(RigidBodyConstraint.PostureChangeConstraintType,robot.getMexModelPtr,joint_ind,lb_change,ub_change,tspan);
-      obj = obj@MultipleTimeLinearPostureConstraint(robot,tspan);
+    function obj = PostureChangeConstraint(robot,joint_ind,lb_change,ub_change)
+      ptr = constructPtrRigidBodyConstraintmex(RigidBodyConstraint.PostureChangeConstraintType,robot.getMexModelPtr,joint_ind,lb_change,ub_change);
+      obj = obj@MultipleTimeLinearPostureConstraint(robot);
       obj = obj.setJointChangeBounds(joint_ind,lb_change,ub_change);
       obj.type = RigidBodyConstraint.PostureChangeConstraintType;
       obj.mex_ptr = ptr;
     end
-    
-    
+
+
     function num = getNumConstraint(obj,t)
       valid_flag = obj.isTimeValid(t);
```

Comment:
remove `t` from input argument, replace it with `num_knots`. The number of constraints only depends on the number of knots, not the exact time of each knot